src/ProxyManager/ProxyGenerator/AccessInterceptorScopeLocalizer/MethodGenerator/MagicGet.php

```diff
@@ -38,7 +38,8 @@ public function __construct(
                 'name',
                 null,
                 null,
-                'returnValue'
+                'returnValue',
+                $originalClass->isInterface() ? $originalClass->getName() : null
```

Comment:
Mutation test score went down: that's because each of these magic methods requires testing for generated code in case of interfaces given

Reply:
I've picked up the branch and am refactoring now.